k8s/upgrades/0.9.0-1.0.0/claim_blockdevices.sh

```diff
@@ -0,0 +1,201 @@
+function claim_blockdevices_sp() {
+    local spc_name=$1
+    local sp_name=$2
+    ## kubectl command get output in below format
+    ## [sparse-37a7de580322f43a sparse-5a92ced3e2ee21 sparse-5e508018b4dd2c8]
+    ## and then converts to below format
+    ## sparse-37a7de580322f43a,sparse-5a92ced3e2ee21,sparse-5e508018b4dd2c8
+    sp_disk_list=$(kubectl get sp $sp_name \
+                   -o jsonpath="{.spec.disks.diskList}" | tr "[]" " ")
+
+    for disk_name in $sp_disk_list; do
+        create_bdc_claim_bd $spc_name $disk_name
+    done
+}
+
+## Output of below command
+## kubectl exec cstor-sparse-d9r7-66cd7b798c-4qjnt -n openebs -c cstor-pool -- zpool list -v -H -P | awk '{print $1}'
+## cstor-49a012ee-8f1a-11e9-8773-54e1ad4a9dd4
+## mirror
+## /var/openebs/sparse/3-ndm-sparse.img
+## /var/openebs/sparse/0-ndm-sparse.img
+## from the above output extracting disk names
+function get_underlying_disks() {
+    local pod_name=$1
+    local pool_type=$2
+    local zpool_disk_list=$(kubectl exec $pod_name -n $ns -c cstor-pool \
+                          -- zpool list -v -H -P | \
+                          awk '{print $1}' | grep -v cstor | grep -v ${map_pool_type[$pool_type]})
+    echo $zpool_disk_list
+}
+
+## claim_blockdevices_csp accepts spc name and csp list as a parameters
+function claim_blockdevices_csp() {
+    local spc_name=$1
+    local csp_list=$2
+    local sp_name=""
+    local pool_pod_name=""
+    local found=0
+    for csp_name in `echo $csp_list | tr ":" " "`; do
+        pool_pod_name=$(kubectl get pod -n $ns \
+                        -l app=cstor-pool,openebs.io/cstor-pool=$csp_name,openebs.io/storage-pool-claim=$spc_name \
+                        -o jsonpath="{.items[0].metadata.name}")
+        rc=$?; if [ $rc -ne 0 ]; then echo "Failed to get pool pod name for csp: $csp_name | Exit Code: $rc"; error_msg; exit 1; fi
+
+        pool_type=$(kubectl get csp $csp_name \
+                    -o jsonpath='{.spec.poolSpec.poolType}')
+        rc=$?; if [ $rc -ne 0 ]; then echo "Failed to get pool type for csp: $csp_name | Exit Code: $rc"; error_msg; exit 1; fi
+
+
+        csp_disk_list=$(kubectl get csp $csp_name \
+                        -o jsonpath='{.spec.disks.diskList}' | \
+                        tr "[]" " ")
+
+        zpool_disk_list=$(get_underlying_disks $pool_pod_name $pool_type)
+
+        ## In some platforms we are getting some suffix to the zpool_disk_list
+        for zpool_disk in $zpool_disk_list; do
+            found=0
+            for csp_disk in $csp_disk_list; do
+                if [[ "$zpool_disk" == "$csp_disk"* ]]; then
+                    found=1
+                    break
+                fi
+            done
+            if [ $found == 0 ]; then
+                echo "zpool disk: $zpool_disk is not found in csp: $csp_name disk list: {$csp_disk_list}"
+                error_msg
+                exit 1
+            fi
+        done
+
+        sp_name=$(kubectl get sp \
```

Comment:
get the number of disks from sp as well and compare the count with csp_disk_list

Reply:
Done